### ISSUE TITLE
Example with a simple microservice

### ISSUE THREAD

**Issue by alejogutierrez** [PoweredLocal/vrata#1]
Hi I was looking to the implementation but still don't know how to interact with a service.
It is possible to see an example with a simple service?
Also a license would be awesome, thanks

**Comment by dusterio** [comment]
I've just added a license - thanks for noticing this! :)

Sure, we can provide an example. Does your simple microservice have Swagger documentation?

**Comment by alejogutierrez** [comment]
Well not really, in fact I found your repo because right now I'm moving from a monolitic Laravel app to microsevices written in Lumen.
So does it mean that you have to use an specific modeling lang like swagger o RAML? 
Thanks again

**Comment by dusterio** [comment]
You can either do it manually by providing your API gateway routes in GATEWAY_ROUTES environment variable, or you could parse Swagger JSON automatically. Swagger is just a nice way

**Comment by alejogutierrez** [comment]
Fine I'm going to take a look using swagger.

**Comment by reindeer77** [comment]
I agree that it would be awesome to have some configuration examples (in my case swagger based microservice, but any further hint would be really appreciated) Thanks!.

**Comment by dusterio** [comment]
Sure, you are right. Will add some examples tomorrow!

**Comment by moon0326** [comment]
@dusterio Hi Dusterio, where you able to add examples? The doc really does not make sense. I'm pretty sure that no one can run the full example, especially Swagger parts. It would be nice to working examples with Swagger support.

**Comment by dusterio** [comment]
@moon0326 I did but perhaps it wasn't enough :) I will review the documentation today and make sure it's comprehensive!

**Comment by moon0326** [comment]
@dusterio Thank you so much!

**Comment by spacecomx** [comment]
@moon0326 Have you managed to use swagger with the api gateway. I don't get how this works. Any help will be appreciated. What does the routes.json file do?

**Comment by dusterio** [comment]
@spacecomx your microservices are expected to expose Swagger documentation endpoints (version 1), all actions will be imported upon start. routes.json is an internal, auto-generated file - a result of parsing of those Swagger endpoints.

**Comment by dusterio** [comment]
@spacecomx I've just checked and Petstore is _only_ provided in Swagger v2, it didn't exist till v2 I suppose. I guess it's time to add Swagger v2 support to Vrata.

The reason I stuck with v1 only originally is - all major frameworks I checked at that time (~6 months ago) only had packages for v1.

**Comment by spacecomx** [comment]
@dusterio Oh you a Star. When creating apis with swagger it defaults to v2.0. Trying to change it to v1 runs an error in the UI. I saw you mentioned v1.0 in your docs, but did not expect that to be an issue at the time. Any time frame on the support for v2? Thanks for the quick responses, wow!

**Comment by dusterio** [comment]
@spacecomx I'll check tomorrow! if it's something simple (the difference is not too big) - I will add a v2 parser tomorrow :)

**Comment by spacecomx** [comment]
Thank your attention and addressing enquiries so quickly. Please can you include an example the petstore setup in the gateway config or docs. Most will look at swagger docs and reference the petstore api to understand the workflow. Look forward to the update to v2.

**Comment by spacecomx** [comment]
@dusterio Just a side note. Swagger 3.0 is planned to be released by the end of July. The name will be changing to OpenAPI, but the tooling will continue to have the Swagger name.

**Comment by spacecomx** [comment]
@dusterio Do you have any feedback on the swagger v2.0 requirements. I was hoping to get the gateway up an running the weekend. If not ready, is there anyway to manually set the api endpoints without swagger so I can get up an running. I have two services that must be pushed into the gateway. Please help, thank you.

**Comment by rizwanmcs** [comment]
where we will define routes mapping in this project?

**Comment by harunnryd** [comment]
@dusterio Hi, where you able to add examples? :(

**Comment by agungsugiarto** [comment]
no example with swagger :(

**Comment by eddycastillon** [comment]
Thank you so much for this awesome gateway.  I am trying to use this one however I have a couple questions about swagger. Until I know there are two ways to generate documentation for an API using swagger.

1 - Using the public swagger editor: https://swagger.io/. It  generates API documentation and lets you share and collaborate  with people. I am able to export my documentation but I do not know how to use  the exported  documentation ( swagger.json ) to discover it. through the gateway.  I think I need to create a controller, route ,etc.

2 - Using some composer packages that generates that documentation. To generate the documentation , basically you need to use the swagger notation in each controller and function. The package exposes your documentation so the gateway can discover the service.

Based on your experience, what is the best approach to document an API.  If you use some composer package for lumen,  Could you share the name please.

Thank you

**Comment by harunnryd** [comment]
@eddycastillon you can use my api gateway bro :) https://github.com/harunnryd/gateid
